v2/log.go

```diff
@@ -0,0 +1,234 @@
+func (l *sLogger) Infof(format string, params ...any) {
+	l.toSlogf(levelInfo, format, params...)
+}
+
+// Warnf formats message according to format specifier, prepends the prefix as
+// necessary, and writes to log with btclog.LevelWarn.
+//
+// This is part of the Logger interface implementation.
+func (l *sLogger) Warnf(format string, params ...any) {
+	l.toSlogf(levelWarn, format, params...)
+}
+
+// Errorf formats message according to format specifier, prepends the prefix as
+// necessary, and writes to log with btclog.LevelError.
+//
+// This is part of the Logger interface implementation.
+func (l *sLogger) Errorf(format string, params ...any) {
+	l.toSlogf(levelError, format, params...)
+}
+
+// Criticalf formats message according to format specifier, prepends the prefix as
+// necessary, and writes to log with btclog.LevelCritical.
+//
+// This is part of the Logger interface implementation.
+func (l *sLogger) Criticalf(format string, params ...any) {
+	l.toSlogf(levelCritical, format, params...)
+}
+
+// Trace formats message using the default formats for its operands, prepends
+// the prefix as necessary, and writes to log with btclog.LevelTrace.
+//
+// This is part of the Logger interface implementation.
+func (l *sLogger) Trace(v ...any) {
+	l.toSlog(levelTrace, v...)
+}
+
+// Debug formats message using the default formats for its operands, prepends
+// the prefix as necessary, and writes to log with btclog.LevelDebug.
+//
+// This is part of the Logger interface implementation.
+func (l *sLogger) Debug(v ...any) {
+	l.toSlog(levelDebug, v...)
+}
+
+// Info formats message using the default formats for its operands, prepends
+// the prefix as necessary, and writes to log with btclog.LevelInfo.
+//
+// This is part of the Logger interface implementation.
+func (l *sLogger) Info(v ...any) {
+	l.toSlog(levelInfo, v...)
+}
+
+// Warn formats message using the default formats for its operands, prepends
+// the prefix as necessary, and writes to log with btclog.LevelWarn.
+//
+// This is part of the Logger interface implementation.
+func (l *sLogger) Warn(v ...any) {
+	l.toSlog(levelWarn, v...)
+}
+
+// Error formats message using the default formats for its operands, prepends
+// the prefix as necessary, and writes to log with btclog.LevelError.
+//
+// This is part of the Logger interface implementation.
+func (l *sLogger) Error(v ...any) {
+	l.toSlog(levelError, v...)
+}
+
+// Critical formats message using the default formats for its operands, prepends
+// the prefix as necessary, and writes to log with btclog.LevelCritical.
+//
+// This is part of the Logger interface implementation.
+func (l *sLogger) Critical(v ...any) {
+	l.toSlog(levelCritical, v...)
+}
+
+// TraceS writes a structured log with the given message and key-value pair
+// attributes with btclog.LevelTrace to the log.
+//
+// This is part of the Logger interface implementation.
+func (l *sLogger) TraceS(ctx context.Context, msg string, attrs ...any) {
```

Comment:
Wanna point out this does create some overhead as we now need to be concerned about context leak? Is `Background` context auto GCed?

Reply:
I think passing a context around in go is generally considered a good pattern. If we are doing lots of context.Background() calls, then it means we should update the parent function to take a context. ie, if we are creating lots of context.Backgrounds() then they really should actually be `context.TODO()`. 

Passing in the context here is really nice cause we can add request scoped attributes that we want to include in logs to the context and then dont need to remember to log them